### PR TITLE
Enable CPU-only builds

### LINKED ISSUE #66: Use BuildKit cache mounts to reduce conda install time in CI

### LINKED ISSUE #129: Provide CPU-only build
Provide build that does not require any GPU-based dependencies (including libcuml) in order to accommodate slimmed-down builds for CPU-only environments

### LINKED ISSUE #148: Consider using Hypothesis to generate input arrays in tests
Using Hypothesis may increase our chances of finding edge cases of the sort that caused problems with categorical feature support before the release of 21.11.

### PR DESCRIPTION
**Pull request by wphicks**
Refactor to conform to CPU-only build pattern established in RAPIDS-Triton. Also introduce Hypothesis for testing as part of refactor of end-to-end tests.

Note that this PR temporarily disables linting in CI. This will be corrected in a separate PR to avoid adding additional unrelated changes to what is already a fairly large revision. Because this PR also sets the stage for ARM builds and clang-format 8 was not available to be installed through conda on ARM, it was necessary to bump the clang-format version, which in turn required an update of the clang-format rules.

Resolve #148
Resolve #129
Resolve #66

### CODE REVIEW

**Review by lowener — COMMENTED** [review]
Just a few comments on CMake. Nothing to say on the C++ side, it looks very good!

File: cmake/thirdparty/get_rapids-triton.cmake [hunk 1]
```diff
@@ -42,5 +42,5 @@ endfunction()
 # CPM_raft_SOURCE=/path/to/local/raft
 find_and_configure_rapids_triton(VERSION    21.10
                         FORK       rapidsai
-                        PINNED_TAG branch-21.10
+                        PINNED_TAG 3a05c3b2c992aed57de2c1ab1e7d0eabf68d1a3c
```
[the remark this answers is not in the document]
Comment (wphicks) [reply cmake/thirdparty/get_rapids-triton.cmake:45]: Yep! As soon as [this PR](https://github.com/rapidsai/rapids-triton/pull/12) is approved.

File: conda/environments/triton_test.yml [hunk 2]
```diff
@@ -5,14 +5,18 @@ channels:
...
   - lightgbm
   - pip
+  - pytest
   - python=3.8
   - rapidsai::py-xgboost
   - scikit-learn
+  - treelite
   - pip:
       - tritonclient[all]
+      - git+https://github.com/rapidsai/rapids-triton.git@03a5977ab1e2fce28d1a0755b668253a2b87e713#subdirectory=python
```
[the remark this answers is not in the document]
Comment (wphicks) [reply conda/environments/triton_test.yml:22]: Also waiting on the same PR, but yep, it sure can!

**Review by lowener — APPROVED** [review]
LGTM
